addons/survey/models/survey_user_input.py

```diff
@@ -623,6 +625,24 @@ def _message_get_suggested_recipients(self):
                 )
         return recipients
 
+    def _notify_new_participation_subscribers(self):
+        subtype_id = self.env.ref('survey.mt_survey_survey_user_input_completed', raise_if_not_found=False)
+        group_survey_user_id = self.env.ref('survey.group_survey_user', raise_if_not_found=False)
+        if not self.ids or not subtype_id or not group_survey_user_id:
+            return
+        # Only post if there are any allowed followers
+        recipients_data = self.env['mail.followers']._get_recipient_data(self.survey_id, 'notification', subtype_id.id)
+        followed_survey_ids = [
+            survey_id
+            for survey_id, followers in recipients_data.items()
+            if any(group_survey_user_id.id in follower['groups'] for follower in followers.values())
+        ]
+        for user_input in self.filtered(lambda user_input_: user_input_.survey_id.id in followed_survey_ids):
+            user_input.message_post(
```

Comment:
Continuing thread on author (as I can't access it anymore): well you don't have to use public user when it is public user, as it does not add information. Generally what we do is: if the user is a real user use its partner as author (aka not public and not in a su environment); otherwise we are in some su / cron / brol and we can use odoobot. As you prefer in the end, just pointing out things we do (which are not necessarily the best but public / su / root is a recurring issue when posting).